src/interface/viewport.c

```diff
@@ -844,6 +846,342 @@ void sub_0x69E8B0(uint32 eax, uint32 ecx){
 	//return;
 }
 
+/*rct2: 0x006C42D9*/
+int sub_6C42D9(rct_string_id string_id, int scroll, int ebp)
+{
+	rct_drawpixelinfo* dpi = RCT2_GLOBAL(0x140E9A8, rct_drawpixelinfo*);
+	if (dpi->zoom_level != 0) return 0x626;
+	RCT2_GLOBAL(0x9D7A80, uint32_t)++;
+	uint32_t edx = 0xFFFFFFFF;
+	for (int i = 0; i < 0x20; i++)
+	{
+		uint8_t* unknown_pointer = RCT2_ADDRESS(0x9C3840, uint8_t) + 0xA12 * i;
+		if (edx >= *((int32_t*)(unknown_pointer + 0x0E)))
+		{
+			edx = *((int32_t*)(unknown_pointer + 0x0E));
+			RCT2_GLOBAL(0x9D7A84, uint32_t) = i;
+			RCT2_GLOBAL(0x9D7A88, uint32_t) = (uint32_t)unknown_pointer;
+		}
+		if (*((rct_string_id*)unknown_pointer) == string_id &&
+			*((uint32_t*)(unknown_pointer + 0x02)) == RCT2_GLOBAL(RCT2_ADDRESS_COMMON_FORMAT_ARGS, uint32_t) &&
+			*((uint32_t*)(unknown_pointer + 0x06)) == RCT2_GLOBAL(0x13CE956, uint32_t) &&
+			*((uint16_t*)(unknown_pointer + 0x0A)) == scroll &&
+			*((uint16_t*)(unknown_pointer + 0x0C)) == ebp)
+		{
+			*((int32_t*)(unknown_pointer + 0x0E)) = RCT2_GLOBAL(0x9D7A80, uint32_t);
+			return i + 0x606;
+		}
+	}
+	uint8_t* unknown_pointer = RCT2_GLOBAL(0x9D7A88, uint8_t*);
+	*((rct_string_id*)unknown_pointer) = string_id;
+	*((uint32_t*)(unknown_pointer + 0x02)) = RCT2_GLOBAL(RCT2_ADDRESS_COMMON_FORMAT_ARGS, uint32_t);
+	*((uint32_t*)(unknown_pointer + 0x06)) = RCT2_GLOBAL(0x13CE956, uint32_t);
+	*((uint16_t*)(unknown_pointer + 0x0A)) = scroll;
+	*((uint16_t*)(unknown_pointer + 0x0C)) = ebp;
+	*((int32_t*)(unknown_pointer + 0x0E)) = RCT2_GLOBAL(0x9D7A80, uint32_t);
+	unknown_pointer += 0x12;
+	memset(unknown_pointer, 0, 0x280 * 4);
+	format_string(RCT2_ADDRESS(RCT2_ADDRESS_COMMON_STRING_FORMAT_BUFFER, char), string_id, (void*)RCT2_ADDRESS_COMMON_FORMAT_ARGS);
+	int al = RCT2_GLOBAL(0x13CE959, uint8_t);
+	int edi = al & 0x7F;
+	int offs = 0;
+	if (al >= 0x80) offs = 2;
+	RCT2_GLOBAL(0x9D7A8C, uint8_t) = RCT2_ADDRESS(0x0141FC47, uint8_t)[offs + edi * 8];
+	int16_t* unk = RCT2_ADDRESS(0x992FB8, uint16_t*)[ebp];
+	uint8_t* format_result = RCT2_ADDRESS(RCT2_ADDRESS_COMMON_STRING_FORMAT_BUFFER, uint8_t);
+	while (true)
+	{
+		al = *format_result;
+		format_result++;
+		if (al == 0)
+		{
+			format_result = RCT2_ADDRESS(RCT2_ADDRESS_COMMON_STRING_FORMAT_BUFFER, char);
+			continue;
+		}
+		if (al < 0x9C && al >= 0x8E)
```

Comment:
See https://github.com/IntelOrca/OpenRCT2/blob/develop/src/drawing/string.c#L1018-L1037
These offsets are known. (FORMAT_COLOUR_CODE_START, FORMAT_COLOUR_CODE_END)
I'm surprised that banners have the ability to change colours.